nucleus/cluster/cli/src/main/java/com/sun/enterprise/admin/cli/cluster/SetupSshKey.java

```diff
@@ -53,65 +57,55 @@ public final class SetupSshKey extends NativeRemoteCommandsBase {
     @Inject
     private ServiceLocator habitat;
 
-    public SetupSshKey() {
-    }
 
-    /**
-     */
     @Override
-    protected void validate()
-            throws CommandException {
+    protected void validate() throws CommandException {
         super.validate();
         Globals.setDefaultHabitat(habitat);
 
         if (sshkeyfile == null) {
-            //if user hasn't specified a key file and there is no key file at default
-            //location, then generate one
-            String existingKey = SSHUtil.getExistingKeyFile();
+            // if user hasn't specified a key file and there is no key file at default
+            // location, then generate one
+            File existingKey = SSHUtil.getExistingKeyFile();
             if (existingKey == null) {
-                sshkeyfile = SSHUtil.getDefaultKeyFile();
+                sshkeyfile = SSHUtil.getDefaultKeyFile().getAbsolutePath();
                 if (promptForKeyGeneration()) {
                     generatekey = true;
                 }
-            }
-            else {
-                //there is a key that requires to be distributed, hence need password
+            } else {
+                // there is a key that requires to be distributed, hence need password
                 promptPass = true;
-                sshkeyfile = existingKey;
-
-                if (SSHUtil.isEncryptedKey(sshkeyfile)) {
+                sshkeyfile = existingKey.getAbsolutePath();
+                if (SSHUtil.isEncryptedKey(existingKey)) {
                     sshkeypassphrase = getSSHPassphrase(false);
                 }
             }
-        }
-        else {
-            promptPass = SSHUtil.validateKeyFile(sshkeyfile);
-            if (SSHUtil.isEncryptedKey(sshkeyfile)) {
+        } else {
+            File keyFile = new File(sshkeyfile);
+            promptPass = SSHUtil.validateKeyFile(keyFile);
+            if (SSHUtil.isEncryptedKey(new File(sshkeyfile))) {
```

Comment:
Isn't it unnecessary to re-new File(sshkeyfile) here since the `keyFile` is created just above?